apps/src/localization/Localization.ts

```diff
@@ -0,0 +1,305 @@
+            | {localeOptions: Omit<LanguageInfo, 'rtl'>[]}
+            | undefined
+        )?.localeOptions || []
+      ).map(({text, value}) => ({
+        text,
+        value,
+        rtl: this.isRTL(value),
+      }));
+    }
+
+    if (this.localeList.length === 0) {
+      // There's no small footer either (or no smallfooter data)
+      // Query the locale dropdown on the page and read from the <option> listings
+      this.localeList = Array.from(
+        document.querySelectorAll('select#locale option')
+      ).map(option => ({
+        text: option.textContent as string,
+        value: (option as HTMLInputElement).value,
+        rtl: this.isRTL((option as HTMLInputElement).value),
+      }));
+    }
+
+    return this.localeList;
+  }
+
+  /**
+   * Registers a callback for the given event.
+   *
+   * @param event - The name of the event to register.
+   * @param callback - The callback to perform when the event is triggered.
+   */
+  on(event: string, callback: TranslationCallback): void {
+    this.callbacks ||= {};
+    this.callbacks[event] ||= [];
+    this.callbacks[event].push(callback);
+
+    if (event === 'change') {
+      // If we aren't in the source language, let's trigger the change event
+      // right away.
+      this.trigger('change', {
+        code: this.locale,
+        rtl: this.rtl,
+      });
+    }
+  }
+
+  /**
+   * Deregisters a callback for the given event.
+   *
+   * @param event - The name of the event to deregister.
+   * @param callback - The callback that was registered.
+   */
+  off(event: string, callback: TranslationCallback): void {
+    this.callbacks ||= {};
+    this.callbacks[event] = (this.callbacks[event] || []).filter(
+      item => item !== callback
+    );
+  }
+
+  /**
+   * Triggers an event with the given data to provide to the event callbacks.
+   *
+   * @param event - The name of the event to trigger.
+   * @param data - The data to pass to the previously registered event callbacks.
+   */
+  trigger(event: string, info: TranslationCallbackData) {
+    const callbacks = this.callbacks[event] || [];
+    for (const callback of callbacks) {
+      callback(info);
+    }
+  }
+
+  /**
+   * This will ensure that the given string is sent to the translation
+   * manager for translation, but does not return the translated string.
+   *
+   * It always returns what it is sent.
+   *
+   * See `translate` for more explanation of labels.
+   *
+   * @param key - The source text, typically English text, or original text.
+   * @param labels - Optional set of categorical labels to attach to the text.
+   */
+  translatable(key: string, labels: string[] = []): string {
+    this.translate(key, labels);
+    return key;
+  }
+
+  /**
+   * Whether or not the given language code represents a right-to-left language.
+   *
+   * @param code - The language code (e.g. 'en-US')
+   */
+  isRTL(code: string): boolean {
+    return ['fa'].includes(code.split('-')[0]);
```

Comment:
Could this be a bit more fault tolerant? Thinking something like having an allow set of country codes as a class field like

```
  this.rtlCountries = new Set(['fa']);
```

```suggestion
    if (!code) {
      return false;
    }
    
    const countryCodeSplit = code.split('-);
    
    if (!Array.isArray(countryCodeSplit) && countryCodeSplit.length !== 1) {
      return false;
    }
    
    const countryCode = countryCodeSplit[0];
        
    return this.rtlCountries.has(countryCode)
```

Reply:
Your suggestion does not cover the case where `code` is undefined or not a string. If `code` is a string, `split` will always return an array of at least one string. The type validation already checks for this.

But, yes I can add a couple of unit tests to cover the expected behavior of Localize and the functional behavior of these accessors.